src/figure_factory.py

```diff
@@ -0,0 +1,19 @@
+from figure import Figure
+from figure_kind import FigureKind
+from point import Point
+
+
+class FigureFactory:
+    kind_figure_points_mapping: dict[FigureKind, list[Point]] = {
+        FigureKind.SQUARE: [Point(0, 0), Point(0, 1), Point(1, 0), Point(1, 1)],
+        FigureKind.LINE: [Point(0, 0), Point(0, 1), Point(0, 2), Point(0, 3)],
+        FigureKind.SNAKE_RIGHT: [Point(0, 1), Point(0, 2), Point(1, 0), Point(1, 1)],
+        FigureKind.SNAKE_LEFT: [Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 2)],
+        FigureKind.CORNER_RIGHT: [Point(0, 0), Point(1, 0), Point(2, 0), Point(2, 1)],
+        FigureKind.CORNER_LEFT: [Point(0, 1), Point(1, 1), Point(2, 0), Point(2, 1)],
+        FigureKind.T_LETTER: [Point(0, 0), Point(0, 1), Point(0, 2), Point(1, 1)],
+    }
+
+    @staticmethod
+    def create(kind: FigureKind) -> Figure:
+        return Figure(kind, Point(0, 0), FigureFactory.kind_figure_points_mapping[kind])
```

Comment:
1) extract `FigureFactory.kind_figure_points_mapping[kind]` to variable e.g. `figure_points`
2) please pay attention: you need to make a deep copy ...
```py
import copy

copy.deepcopy()
```
... of `FigureFactory.kind_figure_points_mapping[kind]`

Otherwise, you change the mapping

Reply:
How I change the mapping just using it?

Reply:
```
a = FigureFactory.create(FigureKind.SQUARE)
a.points = [Point(1, 0)]
print(FigureFactory.kind_figure_points_mapping[FigureKind.SQUARE])  # [Point(x=0, y=0), Point(x=0, y=1), Point(x=1, y=0), Point(x=1, y=1)]
```

Reply:
you had overridden the reference not updating the value

instead of `a.points = [Point(1, 0)]` try `aa.points.append(Point(-27, -15))

Reply:
Ok, done